### ISSUE TITLE
Ugly fail after after entering info about a _datafile by hand

### ISSUE THREAD

**Issue by sfsinger19103** [sfsinger19103/election_anomaly#43]
On 788ba00b, Error message after entering info about a _datafile by hand:

```Database foreign id assigned for each Foreign_Id in ExternalIdentifier.
Pick a record from _datafile list in file system:
Enter info for new _datafile record.
Enter the short_nameEnter the file_nameEnter the file_dateEnter the download_dateEnter the sourceEnter the noteSpecify the ReportingUnit for this _datafile
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/pydevd.py", line 1434, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/singer3/PycharmProjects/results_analysis/src/election_anomaly/main_routines/051_load_datafile_with_aux.py", line 28, in <module>
    ui.track_results_file(d['project_root'],sess,d['results_file'])
  File "/Users/singer3/PycharmProjects/results_analysis/src/election_anomaly/user_interface/__init__.py", line 96, in track_results_file
    sess,project_root,'_datafile',known_info_d={'file_name':filename})
  File "/Users/singer3/PycharmProjects/results_analysis/src/election_anomaly/user_interface/__init__.py", line 275, in pick_or_create_record
    sess,element,known_info_d=known_info_d)
  File "/Users/singer3/PycharmProjects/results_analysis/src/election_anomaly/user_interface/__init__.py", line 517, in get_record_info_from_user
    new[c_plain] = db_record[dbr.get_name_field(fk_df.loc[c,'foreign_table_name'])]
TypeError: 'NoneType' object is not subscriptable
```

**Comment by sfsinger19103** [comment]
NB: there was no Election or ReportingUnit loaded to the db, so maybe that was the problem.

**Comment by ericmtsai** [comment]
This should hopefully be addressed with the work that @Raghu-Srungavarapu is doing. We're reworking the _datafile info a bit. Let's keep this issue around for now and see if it still comes up once that work is merged into the feature branch.

**Comment by ericmtsai** [comment]
I believe we can close this. I can't see a way that we could make this error happen, provided the user enters in the correct params to the `DataLoader.track_results()` function. However, I did create [this](https://github.com/sfsinger19103/election_anomaly/issues/74) as the work in that part of the code is somewhat incomplete. I suggest we close this issue.
